src/lib/statistics/chunk_statistics/min_max_filter.hpp

```diff
@@ -18,7 +18,13 @@ class MinMaxFilter : public AbstractFilter {
 
   bool can_prune(const PredicateCondition predicate_type, const AllTypeVariant& variant_value,
                  const std::optional<AllTypeVariant>& variant_value2 = std::nullopt) const override {
+    // Early exit for NULL variants.
+    if (variant_is_null(variant_value)) {
```

Comment:
Wouldn't the safer (additional) condition be `if predicate_type is IsNull or IsNotNull`? Because that's the operation the the Filter(s) have no idea about, right? variant_value being NULL for these conditions seems to be more of an indirect side-effect.

Reply:
The current approach also ensures that a predicate such as `a = NULL` does not crash.
I could check for the predicate type, but this would be three lines of code instead of the default value in the switch.

But you reminded me that we might want to check for value2 to be non-null as well.

I can live with any solution.

Reply:
But you reminded me that we might want to check for value2 to be non-null as well. But I don't like that to be in any statistics component either.

Just ignore such malformed cases or should that be part of the optimizer / parser?

Reply:
I'd just do `if (is_variant_null(value) or is_variant_null(value2) or predicate_type is IS (NOT) NULL) return false-Filter-dont-know-how-to-handle-this`, I think that's the least-code approach to ensuring that nothing goes wrong.

Reply:
Do you think we should return false for `can_prune(PredicateCondition::LessThan, {5}, NULL_VALUE)` (assume 5 can be pruned)?

Reply:
(Don't worry about it too much, the new CardinalityEstimator will likely wrap PredicateCondition+Value+Value2 into a single data structure and ensure that illegal expressions aren't possible)

Apart from that: yes, I'd say so. If the PredicateCondition is so that value2 can be ignored, then it is valid for the Filter to ignore it, whatever its value is.

Reply:
But if you use the check as I proposed (`if (is_variant_null(value) or is_variant_null(value2) or predicate_type is IS (NOT) NULL)`) and return false, then that would be okay with me as well. It's the callers responsibility to formulate a valid expression.

Reply:
I changed the code accordingly. Ready to merge?